tests/nest_tests/test_gap_junction.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+#
+# test_gap_junction.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import os
+import pytest
+import scipy
+
+import nest
+
+from pynestml.codegeneration.nest_tools import NESTTools
+from pynestml.frontend.pynestml_frontend import generate_nest_target
+
+try:
+    import matplotlib
+    matplotlib.use("Agg")
+    import matplotlib.ticker
+    import matplotlib.pyplot as plt
+    TEST_PLOTS = True
+except Exception:
+    TEST_PLOTS = False
+
+
+@pytest.mark.skipif(NESTTools.detect_nest_version().startswith("v2"),
+                    reason="This test does not support NEST 2")
+class TestGapJunction:
+    r"""Test code generation and perform simulations and numerical checks for gap junction support in linear and non-linear neuron models"""
+
+    @pytest.mark.parametrize("neuron_model", ["iaf_psc_exp", "aeif_cond_exp"])
+    def test_gap_junction_effect_on_membrane_potential(self, neuron_model: str):
+        self.generate_code(neuron_model)
+        for wfr_interpolation_order in [0, 1, 3]:
```

Comment:
Could this be parametrized as well?

Reply:
Nested parametrization like this is the fifth circle of hell. Sadly pytest isn't quite there yet.